### PR TITLE
Improve test coverage for attribution_control.ts

### PR DESCRIPTION
**Pull request by pierreeurope**
## Description
This PR adds comprehensive edge case tests for `AttributionControl` to improve test coverage and ensure robustness.

## New Test Cases
1. **Non-string values filtering** - Verifies that non-string values in `customAttribution` arrays are filtered out
2. **Whitespace-only attributions** - Ensures attributions containing only whitespace are filtered out
3. **Drag behavior** - Tests `_updateCompactMinimize` properly collapses expanded compact attributions during map drag
4. **Cleanup on removal** - Verifies `onRemove` properly cleans up event listeners, DOM elements, and internal state
5. **Empty custom attribution** - Handles empty string in `customAttribution` option
6. **Identical attribution deduplication** - Tests that multiple sources with identical attributions only show once

## Testing
All 37 tests pass, including the 6 new edge case tests.

## Changes
- Added 6 new test cases to `attribution_control.test.ts`
- No changes to production code
- Improves test coverage and documentation of expected behavior

### CODE REVIEW

File: src/ui/control/attribution_control.test.ts [hunk 1]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+        const attribution = new AttributionControl({
+            customAttribution: ['Valid String', 123 as any, null as any, 'Another Valid', undefined as any, '' as any]
+        });
+        map.addControl(attribution);
+
+        // Should only show the valid non-empty strings
+        expect(attribution._innerContainer.innerHTML).toContain('Valid String');
+        expect(attribution._innerContainer.innerHTML).toContain('Another Valid');
+        expect(attribution._innerContainer.innerHTML).not.toContain('123');
+    });
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts]: I don't think we should test things which are invalid accoding to TS.
That is not the public API..


```suggestion
```

File: src/ui/control/attribution_control.test.ts [hunk 2]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+        // Remove the control
+        map.removeControl(attribution);
+        
+        // Verify cleanup
+        expect(container.querySelectorAll('.maplibregl-ctrl-attrib')).toHaveLength(0);
+        expect(attribution._map).toBeUndefined();
+        expect(attribution._compact).toBeUndefined();
+        expect(attribution._attribHTML).toBeUndefined();
+    });
+
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts]: don't think this is a good test..
I mean what does it test (except the non-public API)?

```suggestion
```

File: src/ui/control/attribution_control.test.ts [hunk 3]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+        expect(attribution._container.classList.contains('maplibregl-compact-show')).toBe(false);
+
+        // Click to expand again
+        simulate.click(toggle);
+        expect(attribution._container.classList.contains('maplibregl-compact-show')).toBe(true);
+
+        // Drag should minimize it
+        map.fire('drag');
+        expect(attribution._container.classList.contains('maplibregl-compact-show')).toBe(false);
+    });
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts]: these are two tests. Can we split them?

File: src/ui/control/attribution_control.test.ts [hunk 4]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+            attribution: '\t\n'
+        });
+        map.addSource('valid', {
+            type: 'geojson',
+            data: {type: 'FeatureCollection', features: []},
+            attribution: 'Valid Attribution'
+        });
+        
+        map.addLayer({id: 'w1', type: 'fill', source: 'whitespace1'});
+        map.addLayer({id: 'w2', type: 'fill', source: 'whitespace2'});
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts]: Lets simplify

```suggestion
        map.addSource('whitespace', {
            type: 'geojson',
            data: {type: 'FeatureCollection', features: []},
            attribution: ' '
        });
        
        map.addLayer({id: 'whitespace', type: 'fill', source: 'whitespace'});
```

File: src/ui/control/attribution_control.test.ts [hunk 5]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+
+    test('empty customAttribution string is filtered out', () => {
+        const attribution = new AttributionControl({
+            customAttribution: ''
+        });
+        map.addControl(attribution);
+
+        expect(attribution._innerContainer.innerHTML).toBe('');
+        expect(attribution._container.classList.contains('maplibregl-attrib-empty')).toBe(true);
+    });
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts]: please move the edgecases into their own `describe`

File: src/ui/control/attribution_control.test.ts [hunk 6]
```diff
@@ -566,4 +566,144 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+        map.addLayer({id: 'layer2', type: 'fill', source: 'source2'});
+        map.addLayer({id: 'layer3', type: 'fill', source: 'source3'});
+
+        await sleep(100);
+
+        // Should only show the attribution once
+        const innerHTML = attribution._innerContainer.innerHTML;
+        const count = (innerHTML.match(/Same Attribution/g) || []).length;
+        expect(count).toBe(1);
+    });
```
Comment (CommanderStorm) [inline src/ui/control/attribution_control.test.ts:643]: If something is missing in above `dedupe` test, can we improve that test (possibly break it inito its component parts) instead?

**Comment by pierreeurope** [comment]
Addressed all feedback — thanks @CommanderStorm:
- Removed the non-public-API / invalid TS tests
- Moved edge cases into their own `describe` block
- Simplified the whitespace test to a single source
- Simplified the drag test
- Removed the redundant dedup test (already covered by existing tests)
- Cleaned up the onRemove test

Down from 6 new tests to 4 focused ones.

File: src/ui/control/attribution_control.test.ts [hunk 7]
```diff
@@ -566,4 +566,75 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+
+        test('drag minimizes expanded compact attribution', () => {
+            Object.defineProperty(map.getCanvasContainer(), 'offsetWidth', {value: 600, configurable: true});
+            const attribution = new AttributionControl({
+                compact: true,
+                customAttribution: 'Test Attribution'
+            });
+            map.addControl(attribution);
+
+            // Initially compact-show is present
```
Comment (HarelM) [inline src/ui/control/attribution_control.test.ts]: Remove redundant comments.

Comment (HarelM) [reply src/ui/control/attribution_control.test.ts]: Also, initially compact-show should probably be in a different test to avoid testing too much in this test.

**Comment by HarelM** [comment]
I've added a last comment, looks good otherwise.

File: src/ui/control/attribution_control.test.ts [hunk 8]
```diff
@@ -566,4 +566,82 @@ describe('AttributionControl test regarding the HTML elements details and summar
...
+                data: {type: 'FeatureCollection', features: []},
+                attribution: 'Valid Attribution'
+            });
+
+            map.addLayer({id: 'whitespace', type: 'fill', source: 'whitespace'});
+            map.addLayer({id: 'valid', type: 'fill', source: 'valid'});
+
+            await sleep(100);
+
+            expect(attribution._innerContainer.innerHTML).toBe('Valid Attribution');
```
Comment (HarelM) [inline src/ui/control/attribution_control.test.ts]: These tests assert on the _innerContainer and _container.
It is best not to relay on internal fields.
Please use automation ID or a similar technique to make sure the attribution is as expected.

**Comment by pierreeurope** [comment]
Replaced all internal field access (`_innerContainer`, `_container`, `_map`) with DOM queries via `map.getContainer().querySelector()` and CSS class checks. Also split the compact-show assertion into its own test. Let me know if there's anything else.

**Review by HarelM — APPROVED** [review]
